### PR TITLE
auth endpoints, api, data layer

### PR DESCRIPTION
**Pull request by chaim1221**
so far:
 * auth skeleton
 * fixed auth API endpoints
 * implemented Mongoose

moving to a new PR as there's too much to review already.

### CODE REVIEW

File: node/package.json [hunk 1]
```diff
@@ -18,7 +18,10 @@
   },
   "homepage": "https://github.com/VerbalWebsites/rrk9-store#readme",
   "dependencies": {
+    "bcrypt": "^4.0.1",
```
Comment (chaim1221) [inline node/package.json:21]: mental note uninstall bcrypt as we will not be encrypting paswords

File: node/server.js [hunk 2]
```diff
@@ -19,6 +19,10 @@ router.get('/system/ping', function(request, response) {
   response.json({ message: '200 OK' });
 });
 
+var authController = new AuthController();
```
Comment (chaim1221) [inline node/server.js:22]: const

File: node/server.js [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
+const express = require('express');
+const bodyParser = require('body-parser');
+const AuthController = require('./api/auth/controller.js');
+const LoginController = require('./api/logins.js');
+const ReceiptController = require('./api/receipts.js');
 
 var port = process.env.PORT || 8080;
 
 var app = express();
 var router = express.Router();
```
Comment (chaim1221) [inline node/server.js:10]: const, const, const

File: node/domain/receipts/receipts.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+const Schema = mongoose.Schema;
+
+let ReceiptSchema = new Schema({
+  loginId: [{
+    type: Schema.Types.ObjectId,
+    ref: 'Login'
+  }],
+  email: String,
+  datetime: Date
+});
```
Comment (dearlittlebird) [inline node/domain/receipts/receipts.js:11]: what about the items purchased? is that a different PR?

**Pull request merged**